code/debris/debris.cpp

```diff
@@ -171,11 +169,11 @@ void debris_delete( object * obj )
 	db = &Debris[num];
 
 	Assert( Num_debris_pieces >= 0 );
-	if ( db->is_hull && (db->flags & DEBRIS_EXPIRE) ) {
-		debris_clear_expired_flag(db);
+	if ( db->is_hull && (!db->flags[Debris_Flags::DoNotExpire]) ) {
```

Comment:
You check the DoNotExpire flag both here and inside the `debris_remove_from_hull_list` function.  Want to remove this check?